test/dynamics/perturbation/test_solve_lmde_perturbation.py

```diff
@@ -107,13 +115,15 @@ def test_dyson_like_analytic_case1_1d(self):
         """
 
         def generator(t):
-            return Array([[1, 0], [0, 1]], dtype=complex).data
+            return unp.array([[1, 0], [0, 1]], dtype=complex)
 
         def A0(t):
-            return Array([[0, t], [t**2, 0]], dtype=complex).data
+            return numpy_alias(like=self.array_library()).array(
+                [[0, t], [t**2, 0]], dtype=complex
+            )
 
         def A1(t):
-            return Array([[t, 0], [0, t**2]], dtype=complex).data
+            return unp.array([[t, 0], [0, t**2]], dtype=complex)
```

Comment:
This is the result of the behaviour of how `arraylias` infers libraries when the input is a list, and this impacts JAX compilation. 

In the case of `A1`, the first entry of the nested list is `t`, which when within JAX compilation will be a tracer. In this case, `unp.array` will correctly infer that we want this to be a JAX array. In the case of `A0`, the first entry in the nested list is the float `0.`, and so `unp.array` infers that we want to use a `numpy` array, which causes an error to be raised because `t` is a JAX tracer.

This is a good catch, and I acknowledge that it's subtle and confusing. I didn't change `A1` to use `numpy_alias(like=self.array_library())` out of laziness, but I'll make the change now for consistency/robustness of the tests if these get edited at a later time.